metis-validation/metis-validation-service/src/main/java/eu/europeana/validation/service/SchemaProvider.java

```diff
@@ -149,17 +159,17 @@ private File downloadZipIfNeeded(String zipLocation, String destinationDir)
       throw new SchemaProviderException("Unable to create schemaDirecory");
     }
 
-    final File destinationFile = new File(schemasLocation, ZIP_FILE_NAME);
-    // We know where the schema zips are located: this come from the config files.
-    try (@SuppressWarnings("findsecbugs:URLCONNECTION_SSRF_FD") InputStream urlLocation = new URL(
-            zipLocation).openStream();
-            OutputStream fos = Files.newOutputStream(destinationFile.toPath())
-    ) {
-      IOUtils.copy(urlLocation, fos);
-      return destinationFile;
-    } catch (IOException e) {
-      throw new SchemaProviderException("Unable to store schema file", e);
+    File destinationFile = new File(schemasLocation, ZIP_FILE_NAME);
+    final HttpResponse<Path> httpResponse;
+
+    try {
+      httpResponse = httpClient.send(httpRequest, BodyHandlers.ofFile(destinationFile.toPath()));
+      destinationFile = httpResponse.body().toFile();
+    } catch (IOException | InterruptedException e) {
```

Comment:
If InterruptedException occurs and we are catching it, the interrupted flag should be reset

Reply:
Done